### PR TITLE
Remove mock OTP server from itin existence tests

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [na] The description lists all applicable issues this PR seeks to resolve
- [na] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR removes the use of a mock OTP server for itinerary existence checks. (a mock response getter is used instead).
This may help with E2E tests in #219.

### CODE REVIEW

File: src/main/java/org/opentripplanner/middleware/models/ItineraryExistence.java
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/models/ItineraryExistence.java:1]: Consider removing the `Logger` (line 33) as it is no longer used.

File: src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java [hunk 1]
```diff
@@ -193,6 +195,25 @@ public MonitoredTrip(OtpDispatcherResponse otpDispatcherResponse) throws URISynt
...
+     * Checks that, for each query provided, an itinerary exists.
+     * @return a summary of the itinerary existence results for each day of the week
+     */
+    public boolean checkItineraryExistence(
+        boolean replaceItinerary,
+        Function<OtpRequest, OtpResponse> otpResponseProvider
+    ) throws URISyntaxException {
+        // Get queries to execute by date.
+        List<OtpRequest> queriesByDate = getItineraryExistenceQueries();
+        this.itineraryExistence = new ItineraryExistence(queriesByDate, this.itinerary, this.arriveBy, otpResponseProvider);
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java]: Nit. `this.` are not needed.

File: src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java [hunk 2]
```diff
@@ -193,6 +195,25 @@ public MonitoredTrip(OtpDispatcherResponse otpDispatcherResponse) throws URISynt
...
+        this.itineraryExistence.checkExistence(this);
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java]: Nit. First `this.` is not needed.

File: src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java [hunk 3]
```diff
@@ -193,6 +195,25 @@ public MonitoredTrip(OtpDispatcherResponse otpDispatcherResponse) throws URISynt
...
+        boolean itineraryExists = this.itineraryExistence.allMonitoredDaysAreValid(this);
+        // If itinerary should be replaced, do so if all checked days are valid.
+        return replaceItinerary && itineraryExists
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java:212]: The return value is never used. Also, given that the method is only used in `ItineraryUtilsTest`, should it be moved there instead?

File: src/main/java/org/opentripplanner/middleware/otp/OtpDispatcher.java [hunk 4]
```diff
@@ -120,4 +125,43 @@ private static OtpDispatcherResponse sendOtpRequest(
...
+        return handleOtpDispatcherResponse(() -> sendOtpPlanRequest(OtpVersion.OTP1, otpRequest));
+    }
+
+    private static OtpResponse handleOtpDispatcherResponse(Supplier<OtpDispatcherResponse> otpDispatcherResponseSupplier) {
+        OtpDispatcherResponse otpDispatcherResponse;
+        try {
+            otpDispatcherResponse = otpDispatcherResponseSupplier.get();
+        } catch (Exception e) {
+            BugsnagReporter.reportErrorToBugsnag(
+                "Encountered an error while making a request ot the OTP server.",
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/otp/OtpDispatcher.java]: "to" not "ot".

File: src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java [hunk 5]
```diff
@@ -129,13 +116,13 @@ void canCheckAllItinerariesExist(boolean insertInvalidDay, String message) throw
         mockOtpResponses.set(FRIDAY_INDEX, OtpTestUtils.OTP_DISPATCHER_PLAN_ERROR_RESPONSE.getResponse());
         mockOtpResponses.set(WEDNESDAY_INDEX, OtpTestUtils.OTP_DISPATCHER_PLAN_ERROR_RESPONSE.getResponse());
 
-        OtpTestUtils.setupOtpMocks(mockOtpResponses);
+        MockOtpResponseProvider mockResponses = new MockOtpResponseProvider(mockOtpResponses);
```
Comment (br648) [inline src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java:119]: Perhaps out of scope, but would other classes benefit from this change and aid future testing? Five test classes e.g. `TrackedTripControllerTest` still use `OtpTestUtils.mockOtpServer()`.

File: src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java [hunk 6]
```diff
@@ -165,6 +152,9 @@ void canCheckAllItinerariesExist(boolean insertInvalidDay, String message) throw
         // Days not monitored had an error response, so the check should return invalid for those days.
         assertFalse(existence.wednesday.isValid());
         assertFalse(existence.friday.isValid());
+
+        // Make sure all mpcks were used
```
Comment (br648) [inline src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java]: "mocks" and not "mpcks"

**Review by br648 — CHANGES_REQUESTED** [review]
Some minor changes required. Plus, I'm not sure on scope, other test classes may benefit from this change and aid future testing.

File: src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java (hunk 3 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java:212]: This was an unfinished refactor fixed in https://github.com/ibi-group/otp-middleware/pull/240/commits/df3b4b62278435205497ff8c28f88e9169faf815... Thanks for catching it!

File: src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java (hunk 1 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java]: Updated in https://github.com/ibi-group/otp-middleware/pull/240/commits/df3b4b62278435205497ff8c28f88e9169faf815.

File: src/main/java/org/opentripplanner/middleware/models/ItineraryExistence.java
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/models/ItineraryExistence.java:1]: Updated in https://github.com/ibi-group/otp-middleware/pull/240/commits/146b6c1bb9e1930e0acbcc0caecbd96e9fb913a1.

File: src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java (hunk 6 above)
Comment (binh-dam-ibigroup) [reply src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java]: Thanks, fixed in https://github.com/ibi-group/otp-middleware/pull/240/commits/aadf6e7194991a1d375187325b43f70b6004b780.

File: src/main/java/org/opentripplanner/middleware/otp/OtpDispatcher.java (hunk 4 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/otp/OtpDispatcher.java]: Carry over of a prior typo... fixed in https://github.com/ibi-group/otp-middleware/pull/240/commits/47f339183f5cd76606d0e6836546bc2033ca9d08.

File: src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java (hunk 5 above)
Comment (binh-dam-ibigroup) [reply src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java:119]: I agree. Hopefully we can go one at a time.

**Review by br648 — APPROVED** [review]
All good!

**Review by JymDyerIBI — COMMENTED** [review]
👍  Looks good to me.

**Review by josh-willis-arcadis — APPROVED** [review]
LGTM!

**Pull request merged**